helix-core/src/fuzzy.rs

```diff
@@ -0,0 +1,22 @@
+use nucleo::{CaseMatching, MatcherConfig};
+use parking_lot::Mutex;
+
+pub static MATCHER: Mutex<Option<nucleo::Matcher>> = Mutex::new(None);
```

Comment:
Yeah I like that idea 👍 It felt a little ugly having to reinit the marcher in `helix-temr`

Reply:
I wound up doing something similar but slightly different. I kept the API closer to a mutex (and called the type `LayzMutex`) and used a mapped mutex lock guard instead of a `with` method. That allowed me to avoid nesting a bunch of code into closures.